diff --git a/node_modules/hexo-pangu/lib/filter.js b/node_modules/hexo-pangu/lib/filter.js
index 17cb11e..bb6ab7a 100644
--- a/node_modules/hexo-pangu/lib/filter.js
+++ b/node_modules/hexo-pangu/lib/filter.js
@@ -5,12 +5,17 @@ const { runInNewContext } = require('vm');
 function loadPlugin(file, window) {
   const scriptPath = require.resolve(file);
 
-  // Based on: https://github.com/joyent/node/blob/v0.10.33/src/node.js#L516
   const script = `(function(globalThis, { document, DocumentFragment, NodeFilter, Element, HTMLElement, Text, Node, getComputedStyle }, requestIdleCallback) {
     ${readFileSync(scriptPath)}
   });`;
 
-  const fn = runInNewContext(script);
+  const context = {
+    window,
+    globalThis: window,
+    self: window,
+    console
+  };
+  const fn = runInNewContext(script, context);
   const requestIdleCallback = (callback) => callback({
     didTimeout: false,
     timeRemaining: () => 1e5
